tests/tensor/test_basic_opt.py

```diff
@@ -3569,63 +3570,295 @@ def test_Shape_i_canonicalize():
+        # No optimization on dimshuffle with assert
+        func = function(
+            [self.vec, self.tens],
+            self.alloc_wo_dep.dimshuffle(0, 1, "x") + self.tens,
+            mode=self.fast_compile_mode,
+        )
+        self.verify_op_count(func, 1, Alloc)
+        self.verify_op_count(func, 0, Assert)
+
+        # Optimization on dimshuffle with assert
+        # TODO FIXME: The `BroadcastTo` shapes should use the constants
+        # provided by the first/`Alloc` term, and not the unknown values from
+        # the `tens` term.
+        func = function(
+            [self.vec, self.tens],
+            self.alloc_wo_dep.dimshuffle(0, 1, "x") + self.tens,
+            mode=self.fast_run_mode,
+        )
+        self.verify_op_count(func, 0, Alloc)
+        self.verify_op_count(func, 2, Assert)
+
+        # No optimization on dimshuffle without assert
+        func = function(
+            [self.vec, self.tens],
+            self.alloc_w_dep_tens.dimshuffle(0, 1, "x") + self.tens,
+            mode=self.fast_compile_mode,
+        )
+        self.verify_op_count(func, 1, Alloc)
+        self.verify_op_count(func, 0, Assert)
+
+        # Optimization on dimshuffle without assert
+        func = function(
+            [self.vec, self.tens],
+            self.alloc_w_dep_tens.dimshuffle(0, 1, "x") + self.tens,
+            mode=self.fast_run_mode,
+        )
+        self.verify_op_count(func, 0, Alloc)
+        self.verify_op_count(func, 0, Assert)
+
+    def test_multi_input_single_alloc(self):
+        # No optimization on dimshuffle with assert
+        func = function(
+            [self.vec, self.mat],
+            self.tv_wo_dep + self.tm_wo_dep,
+            mode=self.fast_compile_mode,
+        )
+        self.verify_op_count(func, 2, Alloc)
+        self.verify_op_count(func, 0, Assert)
+
+        # Optimization on dimshuffle with assert
+        # TODO: When we support static shape constraints like `shape[i] != 1`,
+        # reproduce this with such a constraint on `mat` and make sure the
+        # `BroadcastTo` is removed.
+        func = function(
+            [self.vec, self.mat],
+            self.tv_wo_dep + self.tm_wo_dep,
+            mode=self.fast_run_mode,
+        )
+        self.verify_op_count(func, 0, Alloc)
+        self.verify_op_count(func, 0, Assert)
+
+        # No optimization on dimshuffle without assert
+        func = function(
+            [self.vec, self.mat, self.s],
+            self.tv_w_dep + self.tm_w_dep,
+            mode=self.fast_compile_mode,
+        )
+        self.verify_op_count(func, 2, Alloc)
+        self.verify_op_count(func, 0, Assert)
+
+        # Optimization on dimshuffle without assert
+        func = function(
+            [self.vec, self.mat, self.s],
+            self.tv_w_dep + self.tm_w_dep,
+            mode=self.fast_run_mode,
+        )
+        self.verify_op_count(func, 0, Alloc)
+        self.verify_op_count(func, 1, Assert)
 
+    def test_error(self):
```

Comment:
Should this be raising an error? Why was it called like this before?

Reply:
It's just a very poorly named test method.